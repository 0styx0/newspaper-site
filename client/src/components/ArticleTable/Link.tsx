import * as React from 'react';
import { Link } from 'react-router-dom';

interface Props {
    url: string;
    issue: number | string;
}

/**
 * Link to article whose info is given in props
 */
export default function ArticleLink(props: Props) {

    return (
        <Link
            to={`/issue/${props.issue}/story/${encodeURIComponent(props.url)}`}
        >
            {decodeURIComponent(props.url)}
        </Link>
    );
}