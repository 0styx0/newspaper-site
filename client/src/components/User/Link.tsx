import * as React from 'react';
import { Link } from 'react-router-dom';

interface Props {
    fullName: string;
    profileLink: string;
}

/**
 * Link to user
 */
export default function UserLink(props: Props) {

    return (
        <Link
            to={`/u/${props.profileLink}`}
        >
            {props.fullName}
        </Link>
    );
}