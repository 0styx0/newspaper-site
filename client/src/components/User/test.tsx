import * as React from 'react';
import Link from './Link';
import * as renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';

describe('<Link>', () => {

    describe('snapshots', () => {

        it('renders correctly', () => {


            const tree = renderer.create(
                <MemoryRouter>
                    <Link profileLink="hello" fullName="Bob Smith" />
                </MemoryRouter>
            ).toJSON();

            expect(tree).toMatchSnapshot();
        });
    });
});
