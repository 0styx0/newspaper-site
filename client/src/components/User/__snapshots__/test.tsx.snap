// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link> snapshots renders correctly 1`] = `
<a
  href="/u/hello"
  onClick={[Function]}
>
  Bob Smith
</a>
`;
